packages/core/src/typeschema/validation.ts

```diff
@@ -124,8 +133,18 @@ class ResourceValidator implements ResourceVisitor {
     crawlResource(this.rootResource, this, this.schema);
 
     const issues = this.issues;
-    this.issues = []; // Reset issues to allow re-using the validator for other resources
-    if (issues.length > 0) {
+
+    let foundError = false;
+    for (const issue of issues) {
+      if (issue.severity === 'error') {
+        foundError = true;
+      }
+      if (issue.severity === 'warning' && this.options?.logger) {
+        this.options.logger.warn(`Validator warning: ${issue.details?.text}`, { issue });
```

Comment:
Does the logger include the projectId in the resulting log? It's really helpful for tracking down which customers are affected

Reply:
ah, good call.  will add